waiter/src/waiter/service_description.clj

```diff
@@ -205,8 +219,21 @@
   "If the provided allowed-params was invalid, attempt to generate a friendly error message.
    Return nil for unknown error."
   [issue]
-  (when (get issue "allowed-params")
-    "allowed-params must be a vector with non-empty strings."))
+  (when-let [allowed-params-issue (get issue "allowed-params")]
+    (let [length-violation (some #(= "schema.core.Constrained" (some-> % .schema .getClass .getCanonicalName))
+                                 allowed-params-issue)
+          regex-violation (some #(= "java.util.regex.Pattern" (some-> % .schema .getClass .getCanonicalName))
+                                allowed-params-issue)
+          reserved-violation (some #(= "schema.core.Predicate" (some-> % .schema .getClass .getCanonicalName))
+                                   allowed-params-issue)]
+      (str "allowed-params is invalid. "
+           (when length-violation
+             "Individual params may not be empty. ")
+           (when regex-violation
+             "Individual params must be made up of letters, numbers, and hyphens and must start with a letter. ")
+           (when reserved-violation
+             (str "Individual params cannot start with MESOS_, MARATHON_, or PORT and cannot be "
```

Comment:
Should `WAITER_` be reserved?

Reply:
Added `WAITER_` as a reserved environment variable.